presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -106,8 +116,8 @@ public boolean schemaExists(ConnectorSession session, String schema)
         if (tableHandle.getColumns().isPresent()) {
             return tableHandle.getColumns().get();
         }
-        ColumnsCacheKey key = new ColumnsCacheKey(JdbcIdentity.from(session), tableHandle.getSchemaTableName());
-        return get(columnsCache, key, () -> delegate.getColumns(session, tableHandle));
+
+        return get(columnsCache, buildColumnsCacheKey(session, tableHandle.getSchemaTableName()), () -> delegate.getColumns(session, tableHandle));
```

Comment:
Please do not inline key as this approach makes this line do one thing more (build key) in addition to 2 already existing: Get and Load, but which we cannot split.